utils/catalog_production/add_glsn_properties.py

```diff
@@ -0,0 +1,286 @@
+        lsst_mag_array = np.array(lsst_mag_array)
+        lsst_mags.append(lsst_mag_array)
+        mag_30_list.append(mag_array)
+        redshift_list.append(gal_z)
+    
+    mag_30_list = np.array(mag_30_list).T
+    lsst_mags = np.array(lsst_mags).T
+    redshift_list = np.array(redshift_list)
+
+    print('Getting SEDs')
+
+    sed_name, magNorm, av, rv = sed_from_galacticus_mags(mag_30_list, redshift_list,
+                                                         H0, Om0, sed_min_wave, sed_wave_width, lsst_mags)
+
+    return sed_name, magNorm, av, rv
+
+
+if __name__ == "__main__":
+
+    catalog_version = sys.argv[1]
+    glsn_cat = sys.argv[2]
+
+    dc2_df_system = pd.read_hdf(glsn_cat, key='system')
+    dc2_df_image = pd.read_hdf(glsn_cat, key='image')
+
+    catalog = GCRCatalogs.load_catalog('%s_small' % catalog_version)
+
+    load_limited_cat(dc2_df_system, catalog)
+    data_df = pd.read_csv('sn_matching_checkpoint_1.csv')
+    print('Saved Checkpoint 1.')
+    match_up_glsn(dc2_df_system, data_df)
+    get_catalog_mags(catalog)
+    print('Saved Checkpoint 2.')
+
+    data_df = pd.read_csv('sn_matching_checkpoint_2.csv')
+    gcr_glsn_match = np.genfromtxt('gcr_glsn_match.txt')
+
+    sed_name, magNorm, av, rv = get_30_band_mags(gcr_glsn_match, data_df, catalog)
+
+    sed_name_array = np.array(sed_name)
+    magNorm_array = np.array(magNorm)
+    av_array = np.array(av)
+    rv_array = np.array(rv)
+
+    np.savetxt('sed_name_list_sn.txt', sed_name_array, fmt='%s')
+    np.savetxt('magNorm_list_sn.txt', magNorm_array)
+    np.savetxt('av_list_sn.txt', av_array)
+    np.savetxt('rv_list_sn.txt', rv_array)
+
+    test_bandpassDict = BandpassDict.loadTotalBandpassesFromFiles()
+    imsimband = Bandpass()
+    imsimband.imsimBandpass()
+
+    mag_norm_glsne = []
+    for i, idx in list(enumerate(keep_rows)):
+        if i % 10000 == 0:
+            print(i, idx)
+        test_sed = Sed()
+        test_sed.readSED_flambda(os.path.join(str(os.environ['SIMS_SED_LIBRARY_DIR']), sed_name_array[i]))
+        fnorm = test_sed.calcFluxNorm(dc2_df_system['lensgal_mi'].iloc[idx], test_bandpassDict['i'])
+        test_sed.multiplyFluxNorm(fnorm)
+        magNorm_diff = magNorm_array[3, i] - test_sed.calcMag(imsimband)
+        mag_norm_glsne.append(magNorm_array[:,i] - magNorm_diff)
+
+    mag_norm_glsne = np.array(mag_norm_glsne)
+
+    results_dict = {}
+    print(len(keep_rows), len(gcr_glsn_match[:, 0]))
+    for keep_idx in range(len(keep_rows)):
+        results_dict[str(dc2_df_system['sysno'].iloc[keep_rows[keep_idx]])] = {'z':gcr_glsn_match[:, 0][keep_idx],
+                                                                               'sed_name':sed_name_array[keep_idx],
+                                                                               'magNorm':mag_norm_glsne[keep_idx]}
+
+    keep_systems = dc2_df_system['sysno'].iloc[keep_rows].values
+
+    final_df_z = []
+    final_df_lens_sed = []
```

Comment:
The `Av` and `Rv` values returned by `sed_from_galacticus_mags` is also an important part of the color fitting.  Should we be storing those in the Twinkles caches as well?